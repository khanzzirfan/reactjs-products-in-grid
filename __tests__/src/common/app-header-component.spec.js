import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import AppHeaderComponent from 'src/common/app-header-component';

describe("app header component test", () => {

    let wrapper = null;
    beforeEach(() => {
        wrapper = shallow(<AppHeaderComponent />);
    });

    it("component should be defined", () => {
        expect(wrapper).toBeDefined();
    });

    it("component should match to snapshot", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it("it should render with correct class names", ()=>{
       
        expect(wrapper.find('.text-center').exists()).toBeTruthy();
       
    });

    it("it should render with correct text", ()=>{
        expect(wrapper.text()).toContain("Creatella");
        expect(wrapper.text()).toContain("Products grid");
        expect(wrapper.text()).toContain("This is an ecommerce site");
        
    });

});