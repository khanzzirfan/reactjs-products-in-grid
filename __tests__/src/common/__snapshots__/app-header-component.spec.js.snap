// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app header component test component should match to snapshot 1`] = `
<div
  className="col-md-12 col-lg-12"
>
  <h3
    className="text-center"
  >
    Creatella
    <small>
      Products grid
    </small>
  </h3>
  <span>
    This is an ecommerce site, where you can buy all sorts of ascii faces like \`(ノ・∀・)ノ\` and \`¯_(ツ)_/¯\`, in a wide variety of font sizes.    
  </span>
</div>
`;
